### DISCUSSION TITLE
hi，About itwins embedding other front-end frameworks

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#1065] [Q&A] [answered]
hi.After embedding itwins into other front-end frameworks, itwins is not as smooth as before. When encountering this situation, is there an optimization method? thanks.

Now when I zoomed in on the model, this error appeared in the console: 
WebAppRpcRequest.ts:287 POST http://114.116.240.10:3001/ninezone-sample-app/v1.0/mode/1/context/00f217b5-d617-4e40-bc89-73b1ef1ebaed/imodel/f5c4be45-56d8-4066-8ca6-95bd7728c833/changeset/5000db463b3cd59c088bf9eec32dfcafffb6f954/IModelTileRpcInterface-2.2.0-getTileCacheContainerUrl 500 (Internal Server Error)

Error   |imodeljs-frontend.RpcInterface| Error: Cannot read property 'obtainContainerUrl' of undefined {"message":"Cannot read property 'obtainContainerUrl' of undefined","isError":true,"name":"TypeError","ExceptionType":"BackendError","ActivityId":"e117d76a-b392-45a4-9100-4690b5a144a7","SessionId":"aeece011-bd99-4deb-af6c-fc88e186c1ce","ApplicationId":"2686","ApplicationVersion":"1.0.0"}

**Comment by pmconne** [comment, marked as answer]
Please be more specific about what "other front-end frameworks" means. Something other than React? If so, what specifically?

Please be more specific about "itwins is not as smooth as before". Are you talking about framerate in a viewport? Responsiveness of the UI? Something else?

**Reply by shenshengyi** [reply]
hi, other front-end frameworks, I mean umijs, which is also based on react.
"itwins is not as smooth as before". I mean that view the model scene has become lagging, not as smooth as before.
It is now very suspect that this front-end framework hijacked any information, but there is no direct evidence. 
I mean the experience of previewing the model in the viewport. It has nothing to do with the user interface response. thanks.

**Reply by pmconne** [reply]
Try using an extremely simple model, or turning off all of the models in the view, to see if the viewport is still laggy.
Other than that, given the only difference you mention is this additional UI framework, you'll probably have to use the profiler to find the cause.

**Reply by shenshengyi** [reply]
thank you so much for your suggestion.
Now we plan to separate itwins into a separate react program to complete the current requirements, because it is impossible to determine whether it is a problem with this front-end framework in a short time.
